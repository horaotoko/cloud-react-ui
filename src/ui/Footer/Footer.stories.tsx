import styled from "styled-components";
import { boolean } from "@storybook/addon-knobs";
import { Footer as FooterComponent } from "./Footer";
import { Link, Text } from "../../components";

export default {
  component: FooterComponent,
  title: "ui/Footer",
};

const StyledText = styled(Text)`
  margin-right: 32px;
  letter-spacing: 0.4px;
`;

export const Footer: React.FC = () => {
  return (
    <FooterComponent fixed={boolean("fixed", false)}>
      <StyledText size="small">
        <Link href="#">利用規約</Link>
      </StyledText>

      <StyledText size="small">
        <Link href="#">個人情報保護方針</Link>
      </StyledText>

      <StyledText size="small">&copy; Money Forward, Inc.</StyledText>
    </FooterComponent>
  );
};
