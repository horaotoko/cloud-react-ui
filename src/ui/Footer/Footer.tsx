import React, { useMemo } from 'react';
import styled, { css } from 'styled-components';
import { defaultProps } from '../../theme';

export type Props = {
  fixed?: boolean;
  className?: string;
  children: React.ReactNode;
};

const FixedWrapper = styled.div`
  ${({ theme: { footer } }) => css`
    position: fixed;
    bottom: 0;
    left: 0;
    width: 100%;
    z-index: ${footer.zIndex};
  `}
`;
FixedWrapper.defaultProps = defaultProps;

const StyledFooter = styled.footer`
  ${({ theme: { footer } }) => css`
    display: flex;
    align-items: center;
    justify-content: flex-start;
    width: 100%;
    height: ${footer.height};
    padding: ${footer.padding};
    border-top: ${footer.borderTop};
    background-color: ${footer.backgroundColor};
    box-sizing: border-box;
  `}
`;
StyledFooter.defaultProps = defaultProps;

export const Footer = ({ fixed = false, ...rest }: Props): JSX.Element => {
  const Footer = useMemo(() => <StyledFooter {...rest} />, [rest]);

  return fixed ? <FixedWrapper>{Footer}</FixedWrapper> : Footer;
};
