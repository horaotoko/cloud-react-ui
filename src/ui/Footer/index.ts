export { FooterContainer as Footer } from "./FooterContainer";
export type { FooterProps } from "./FooterContainer";
