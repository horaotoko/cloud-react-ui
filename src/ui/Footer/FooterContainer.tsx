import { FooterPresenter } from "./FooterPresenter";

export type FooterProps = {
  children: React.ReactNode;
};

export function FooterContainer(props: FooterProps): React.ReactElement {
  return <FooterPresenter {...props} />;
}
