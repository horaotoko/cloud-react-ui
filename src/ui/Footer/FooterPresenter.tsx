import React from "react";
import styled from "styled-components";
import { FooterProps } from "./FooterContainer";
import { defaultProps } from "@theme";

const StyledFooter = styled.footer`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: 100%;
  height: ${(props) => props.theme.footer.height};
  padding: ${(props) => props.theme.footer.padding};
  border-top: ${(props) => props.theme.footer.borderTop};
  background-color: ${(props) => props.theme.footer.backgroundColor};
`;
StyledFooter.defaultProps = defaultProps;

export function FooterPresenter({ children }: FooterProps): React.ReactElement {
  return <StyledFooter>{children}</StyledFooter>;
}
