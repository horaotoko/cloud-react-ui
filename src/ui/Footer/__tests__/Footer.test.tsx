import { render } from '@testing-library/react';
import { Footer } from '../Footer';

describe('Footer', () => {
  it('snapshot test', () => {
    const { asFragment } = render(<Footer>Footer</Footer>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('snapshot test (fixed)', () => {
    const { asFragment } = render(<Footer fixed>Footer</Footer>);
    expect(asFragment()).toMatchSnapshot();
  });
});
