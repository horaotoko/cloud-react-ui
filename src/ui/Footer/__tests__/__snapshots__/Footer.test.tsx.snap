// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot test (fixed) 1`] = `
<DocumentFragment>
  .c0 {
  position: fixed;
  bottom: 0;
  left: 0;
  width: 100%;
  z-index: 100;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
  height: 35px;
  padding: 9px 20px 10px;
  border-top: 1px solid #D4D8DD;
  background-color: #EFF1F4;
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <footer
      class="c1"
    >
      Footer
    </footer>
  </div>
</DocumentFragment>
`;

exports[`Footer snapshot test 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
  height: 35px;
  padding: 9px 20px 10px;
  border-top: 1px solid #D4D8DD;
  background-color: #EFF1F4;
  box-sizing: border-box;
}

<footer
    class="c0"
  >
    Footer
  </footer>
</DocumentFragment>
`;
